pennylane/transforms/compile.py

```diff
@@ -62,14 +62,36 @@ def compile(
             for tape expansion into the basis gates.
 
     Returns:
-        pennylane.QNode or qfunc or tuple[List[.QuantumTape], function]: If a QNode is passed,
-        it returns a QNode with the transform added to its transform program.
-        If a tape is passed, returns a tuple containing a list of
-        quantum tapes to be evaluated, and a function to be applied to these
-        tape executions.
+        qnode (QNode) or quantum function (Callable) or tuple[List[QuantumTape], function]: The compiled circuit. The output type is explained in :func:`qml.transform <pennylane.transform>`.
 
     **Example**
 
+    >>> dev = qml.device('default.qubit', wires=[0, 1, 2])
+
+    You can apply the transform directly on a :class:`QNode`:
+
+    .. code-block:: python
+
+        @compile
+        @qml.qnode(device=dev)
+        def circuit(x, y, z):
+            qml.Hadamard(wires=0)
+            qml.Hadamard(wires=1)
+            qml.Hadamard(wires=2)
+            qml.RZ(z, wires=2)
+            qml.CNOT(wires=[2, 1])
+            qml.RX(z, wires=0)
+            qml.CNOT(wires=[1, 0])
+            qml.RX(x, wires=0)
+            qml.CNOT(wires=[1, 0])
+            qml.RZ(-z, wires=2)
+            qml.RX(y, wires=2)
+            qml.PauliY(wires=2)
+            qml.CY(wires=[1, 2])
+            return qml.expval(qml.PauliZ(wires=0))
+
+    The default compilation pipeline is applied before execution.
```

Comment:
@rmoyard do you think this bit is needed? It feels a bit repetitive compared to the same function below.

Reply:
Resolved in conversation - we decided to put longer examples in a usage details section.